statemachine/src/test/java/com/amazonaws/stepfunctions/cloudformation/statemachine/CreateHandlerTest.java

```diff
@@ -268,6 +283,86 @@ public void testDefinitionFromS3InYAML() throws Exception {
         assertThat(response.getResourceModel().getDefinitionString()).isEqualTo(formattedJson);
     }
 
+    @Test
+    public void testNonStringDefinitionSubstitutionFromS3Yaml() throws Exception {
+        String formattedJson = "{\n" +
+                "  \"StartAt\" : \"DummyState\",\n" +
+                "  \"TimeoutSeconds\" : 60,\n" +
+                "  \"States\" : {\n" +
+                "    \"DummyState\" : {\n" +
+                "      \"Type\" : \"Pass\",\n" +
+                "      \"End\" : true\n" +
+                "    }\n" +
+                "  }\n" +
+                "}";
+
+        Map<String, Object> substitutions = new HashMap<>();
+        substitutions.put("startState", "DummyState");
+        substitutions.put("type", "Pass");
+        substitutions.put("timeoutSeconds", 60);
+        substitutions.put("isEnd", true);
+
+        request.getDesiredResourceState().setDefinitionSubstitutions(substitutions);
+        request.getDesiredResourceState().setDefinitionS3Location(new S3Location(DEFAULT_S3_BUCKET, DEFAULT_S3_KEY, DEFAULT_S3_OBJECT_VERSION));
+
+        S3Object s3Object = new S3Object();
+        String yamlDefinitionInS3 = "StartAt: \"${startState}\"\nTimeoutSeconds: ${timeoutSeconds}\nStates:\n  DummyState:\n    Type: Pass\n    End: ${isEnd}\n  ";
+        s3Object.setObjectContent(new StringInputStream(yamlDefinitionInS3));
+        GetObjectResult getObjectResult = new GetObjectResult(s3Object);
+
+        CreateStateMachineResult createStateMachineResult = new CreateStateMachineResult();
+        createStateMachineResult.setStateMachineArn(STATE_MACHINE_ARN);
+
+        Mockito.when(proxy.injectCredentialsAndInvoke(Mockito.any(), Mockito.any(Function.class))).thenReturn(getObjectResult, createStateMachineResult);
+
+        ProgressEvent<ResourceModel, CallbackContext> response
+                = handler.handleRequest(proxy, request, null, logger);
+
+        assertThat(response).isNotNull();
+        assertThat(response.getStatus()).isEqualTo(OperationStatus.SUCCESS);
+        assertThat(response.getResourceModel().getDefinitionString()).isEqualTo(formattedJson);
+    }
+
+    @Test
+    public void testNonStringDefinitionSubstitutionFromS3Json() throws Exception {
```

Comment:
Same as above. Lets drop the 'NonString' part here.

Reply:
will remove it in my next revision